packages/shared/stub-components-validate.js

```diff
@@ -0,0 +1,58 @@
+// @flow
+
+import { camelize, capitalize, hyphenate, throwError } from './util'
+import { compileToFunctions } from 'vue-template-compiler'
+
+export function validateStubOptions (stubOptions: Array<string> | Object, components: Object) {
+  if (Array.isArray(stubOptions)) {
+    if (containsNonStringItem(stubOptions)) {
+      throwError('each item in an options.stubs array must be a string')
+    }
+  } else {
+    if (containsInvalidOptions(stubOptions)) {
+      throwError('options.stub values must be passed a string or component')
+    }
+
+    if (necessaryCompileToFunctionsMissed(stubOptions)) {
+      throwError('vueTemplateCompiler is undefined, you must pass components explicitly if vue-template-compiler is undefined')
+    }
+
+    if (containsTemplateWithCircularReference(stubOptions, components)) {
```

Comment:
This is actually causing bugs for some users. The check is not strict enough, so I've removed the check in the source code for the moment. Could you either remove this code, or improve the code to handle edge cases like: `UnreadState: '<div data-stub-unread-state v-bind="$props"><slot/></div>'`